exporter/kinesisexporter/producer.go

```diff
@@ -0,0 +1,81 @@
+package kinesisexporter
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kinesis"
+	omnition "github.com/signalfx/omnition-kinesis-producer"
+	"github.com/signalfx/omnition-kinesis-producer/loggers/kpzap"
+	"go.uber.org/zap"
+	"time"
+)
+
+// producer provides the interface for a kinesis producer. The producer
+// implementation abstracts the interaction with the kinesis producer library
+// used from the exporter
+type producer interface {
+	start()
```

Comment:
Can these return errors? :please:?

Reply:
What's the case for returning one? Currently it is impossible for `start()` or `stop()` to return an error in our implementation.

Reply:
Aaa, that isn't fun :\

Reply:
@Excaleo , a good error to returned would be something like `ErrAlreadyStarted` to avoid starting the producer several times here.